plugin.go

```diff
@@ -28,40 +45,259 @@ func (p *Plugin) PluginInfo() launchr.PluginInfo {
+				// Kill process in case of timeout
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					for {
+						select {
+						case <-time.After(1 * time.Second):
+							runInfo, errRi := server.GetRunInfo(pluginTmpDir)
+							if errRi != nil {
+								continue
+							}
+
+							cli.Println("Web running in background with PID:%d", pid)
+							if runInfo.BaseURL != "" {
+								cli.Println(runInfo.BaseURL)
+							}
+							return
+						case <-ctx.Done():
+							return
+						}
+					}
+				}()
+
+				wg.Wait()
+
+				// Failed to retrieve run info for a reasonable time
+				if errCtx := ctx.Err(); errCtx != nil {
+					// Kill existing process
+					_ = killProcess(pid)
+
+					// Cleanup temp folder
+					err = os.RemoveAll(pluginTmpDir)
+					if err != nil {
+						log.Debug(err.Error())
+					}
+
+					return errors.New("couldn't start background process")
+				}
+
+				return nil
+			}
+
 			return server.Run(cmd.Context(), p.app, runOpts)
 		},
 	}
-	cmd.Flags().StringVarP(&port, "port", "p", "8080", `Web server port`)
+
+	cmd.Flags().IntVarP(&port, "port", "p", 8080, `Web server port`)
 	cmd.Flags().BoolVarP(&useSwaggerUI, "swagger-ui", "", false, `Serve swagger.json on /api/swagger.json and Swagger UI on /api/swagger-ui`)
+	cmd.Flags().BoolVarP(&foreground, "foreground", "", false, `Run server as foreground process`)
 	cmd.Flags().StringVarP(&proxyClient, "proxy-client", "", "", `Proxies to client web server, useful in local development`)
 	// Command flags.
 	rootCmd.AddCommand(cmd)
 	return nil
 }
+
+func (p *Plugin) runAsBackgroundCmd(cmd *cobra.Command, pidFile string) (int, error) {
+	err := os.MkdirAll(filepath.Dir(pidFile), 0750)
+	if err != nil {
+		return 0, fmt.Errorf("not possible to create tmp directory for %s", pidFile)
+	}
+
+	// Prepare the command to restart itself in the background
+	args := append([]string{cmd.Name()}, os.Args[2:]...)
+
+	command := exec.Command(os.Args[0], args...) //nolint G204
+	command.Env = append(os.Environ(), fmt.Sprintf("%s=1", cobraBackgroundEnvVar))
+
+	// Set platform-specific process ID
+	setSysProcAttr(command)
+
+	err = command.Start()
+	if err != nil {
+		cmd.Println("Failed to start in background:", err)
+		return 0, err
+	}
+
+	err = os.WriteFile(pidFile, []byte(fmt.Sprintf("%d", command.Process.Pid)), os.FileMode(0644))
+	if err != nil {
+		cmd.Println("Failed to write PID file:", err)
+		return 0, err
+	}
+
+	return command.Process.Pid, nil
+}
+
+func (p *Plugin) getPluginTempDir() string {
+	return p.cfg.Path(pluginName)
+}
+
+func isBackGroundEnv() bool {
+	return len(os.Getenv(cobraBackgroundEnvVar)) == 1
+}
+
+func redirectOutputs(folder string) error {
+	err := os.MkdirAll(folder, 0750)
+	if err != nil {
+		return fmt.Errorf("can't create plugin temporary directory")
+	}
+
+	outLog, err := os.Create(fmt.Sprintf("%s/out.log", folder))
```

Comment:
Please, add a todo here to rework logging. I'm working on a global refactoring of logging solution in the application. I must consider this case.